src/Symfony/Component/JsonEncoder/Tests/Decode/LazyInstantiatorTest.php

```diff
@@ -32,17 +33,46 @@ protected function setUp(): void
         }
     }
 
-    public function testCreateLazyGhost()
+    /**
+     * @requires PHP < 8.4
+     */
+    public function testCreateLazyGhostUsingVarExporter()
     {
-        $ghost = (new LazyInstantiator($this->lazyGhostsDir))->instantiate(ClassicDummy::class, []);
+        $ghost = (new LazyInstantiator($this->lazyGhostsDir))->instantiate(ClassicDummy::class, function (ClassicDummy $object): void {
+            $object->id = 123;
+        });
 
-        $this->assertArrayHasKey(\sprintf("\0%sGhost\0lazyObjectState", preg_replace('/\\\\/', '', ClassicDummy::class)), (array) $ghost);
+        $this->assertSame(123, $ghost->id);
     }
 
+    /**
+     * @requires PHP < 8.4
+     */
     public function testCreateCacheFile()
     {
-        (new LazyInstantiator($this->lazyGhostsDir))->instantiate(DummyWithNormalizerAttributes::class, []);
+        (new LazyInstantiator($this->lazyGhostsDir))->instantiate(DummyWithNormalizerAttributes::class, function (ClassicDummy $object): void {});
 
         $this->assertCount(1, glob($this->lazyGhostsDir.'/*'));
     }
+
+    /**
+     * @requires PHP < 8.4
+     */
+    public function testThrowIfLazyGhostDirNotDefined()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        new LazyInstantiator();
+    }
+
+    /**
+     * @requires PHP >= 8.4
```

Comment:
Other occurrences simply use:

```suggestion
     * @requires PHP 8.4
```